collector/lib/ConnScraper.cpp

```diff
@@ -190,6 +190,60 @@ int ReadHexBytes(const char* p, const char* endp, void* buf, int chunk_size, int
   return i;
 }
 
+struct ProcessInfo {
+  std::string comm;
+  std::string exe;
+  std::string exe_path;
+  std::string args;
+};
+
+static void ReadProcessInfo(const char* process_id, int dirfd, struct ProcessInfo& info) {
+  char buffer[PATH_MAX];
+
+  // parse xx/exe
+  ssize_t nread = readlinkat(dirfd, "exe", buffer, sizeof(buffer));
+  if (nread > 0 && nread < ssizeof(buffer)) {
+    buffer[nread] = '\0';
+
+    info.comm = info.exe_path = buffer;
+
+    if (buffer[0] == '/')
+      info.comm = strrchr(buffer, '/') + 1;
+  } else {
+    CLOG(ERROR) << "Could not read 'exe' for " << process_id << ": " << StrError();
+  }
+
+  // parse xx/cmdline
```

Comment:
This could be split into two functions.

Reply:
I agree. This will be moved to to the Process factory soon, I will change that during the transition.